Gemfile

```diff
@@ -4,3 +4,5 @@ gemspec
 gem "jekyll-github-metadata", ">= 2.15"
 
 gem "webrick", "~> 1.7"
+
+gem "html-proofer", "~> 5.0", :group => :development
```

Comment:
There should be a change in the lockfile?

Reply:
Sorry, a bit confused - what lockfile? We haven't merged #1112 yet.

Reply:
> We haven't merged #1112 yet.

Wouldn't it be good to do that a.s.a.p.? I agree that a 2nd review is needed – especially in view of my low familiarity with the workings of Bundler, Jekyll, and gem files!

BTW, the GH Pages version of `just-the-docs-tests` is now built with Jekyll 4, but its `Gemfile` supports local building with Jekyll 3 or 4. (It doesn't need the legacy `github-pages` gem, so I never tried to support that.)

Reply:
Adding a gem to the Gemfile and running bundle install on your machine causes a change to the lockfile. These should be committed together with the change to the Gemfile. Otherwise each and every pipeline run (without cache) will detect the difference between these 2 files, install the locked dependencies from the lockfile and any current version (respecting the constraints) of the new gem.

The `Gemfile` just contains a list of gems and constraints. The `Gemfile.lock` contains a resolved dependency tree. `bundle install` will take the resolved tree from the lockfile and make sure the `Gemfile` hasn't changed.

Reply:
Oh well, nevermind. We should add the lockfile asap.

Reply:
Definitely agree that we should commit a lockfile (and ideally, lockfiles for our test fixtures as well); I think #1112 should be the avenue for that (I can handle merging these in whatever order we'd prefer).

Reply:
See also [my reference to the updated Jekyll recommendation](https://github.com/just-the-docs/just-the-docs/pull/1112#discussion_r1063390196).

Reply:
Let's get #1307 merged (yes, new one), then rebase this PR.

Reply:
Hm, #1307 is requiring some back and forth. To better understand - why is it necessary to merge in that *before* this PR, and why not #1112? From my perspective, I don't see a direct issue that interferes with the purpose of this PR. 

For what it's worth: after this PR, I'll cut a `v0.5.x` and move forward on `v0.6` with the fixed navigation.